### DISCUSSION TITLE
how to apply Pixi.Text texture to the face of a cube?

### DISCUSSION THREAD

**Discussion by ChristieLukose** [jnsmalm/pixi3d#146] [Q&A] [answered]
Hello,
Request you to help me with an issue.
I created a cube using PIXI3D.Mesh3D.createCube() and wanted to apply the PIXI.text texture to one of the face of the cube.
I am unable to figure out how to retrieve the face information of the cube and apply the PIXI.text Texture to it.

Thanks in advance.

**Comment by jnsmalm** [comment, marked as answer]
Hi!

The easiest way to achieve this is to have a cube mesh and a separate face with the text texture (you can use Sprite3D for this).

```javascript
let box = app.stage.addChild(Mesh3D.createCube())

let sprite = box.addChild(new Sprite3D())
sprite.z = 1.01 // The size of the box plus a small amount to avoid z-fighting.

let text = new Text("This is regular\nPixiJS text")
text.updateText(false)

sprite.texture = text.texture
``` 

<img width="473" alt="Screenshot 2022-12-21 at 21 17 53" src="https://user-images.githubusercontent.com/10713751/208995562-fbc211f5-803f-4ccb-b162-2d99d848dcd6.png">

**Reply by ChristieLukose** [reply]
Thank u for your help.
